pkg/proxy/proxy.go

```diff
@@ -19,28 +19,73 @@ import (
 	"github.com/redhat-appstudio/sprayproxy/pkg/metrics"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"golang.org/x/exp/slices"
 )
 
-type BackendsFunc func() []string
+type backend struct {
+	URL string `json:"url"`
+}
 
 type SprayProxy struct {
-	backends    BackendsFunc
-	insecureTLS bool
-	logger      *zap.Logger
+	backends       []string
+	insecureTLS    bool
+	randomBackends bool
+	logger         *zap.Logger
 }
 
-func NewSprayProxy(insecureTLS bool, logger *zap.Logger, backends ...string) (*SprayProxy, error) {
-	backendFn := func() []string {
-		return backends
-	}
-
+func NewSprayProxy(insecureTLS bool, randomBackends bool, logger *zap.Logger, backends ...string) (*SprayProxy, error) {
 	return &SprayProxy{
-		backends:    backendFn,
-		insecureTLS: insecureTLS,
-		logger:      logger,
+		backends:       backends,
+		insecureTLS:    insecureTLS,
+		randomBackends: randomBackends,
+		logger:         logger,
 	}, nil
 }
 
+func (p *SprayProxy) RegisterBackends(c *gin.Context) {
+	if p.randomBackends {
+		var newUrl backend
+		if err := c.ShouldBindJSON(&newUrl); err != nil {
+			p.logger.Info("backend server register request to proxy is rejected")
+			return
+		}
+		if !slices.Contains(p.backends, newUrl.URL) {
+			p.backends = append(p.backends, newUrl.URL)
+			c.String(http.StatusOK, "registered the backend server")
+			p.logger.Info("server registered")
```

Comment:
I think it would be helpful to log the backend URL as well. Not just here but the other similar instances.

Reply:
Wouldn't that expose the urls in the ci logs?

Reply:
Done

Reply:
The URLs are already exposed in the logs. We already log all requests including the forwarding ones and the backend URL is already there.